draft-ietf-lamps-cms-kyber.md

```diff
@@ -27,474 +38,358 @@ author:
 
-- KEM, a Key Encapsulation Mechanism;
-- KDF, a Key Derivation Function, which derives keying data of a specified length from a shared secret value;
-- WRAP, a symmetric key-wrapping scheme, which encrypts keying Data using a key-encrypting key (KEK).
+The KEM functions defined above correspond to the following functions in {{FIPS203}}:
 
-### KEM
+> KeyGen(): ML-KEM.KeyGen() from section 6.1.
 
-A KEM is a cryptographic algorithm consisting of three functions :
+> Encapsulate(): ML-KEM.Encaps() from section 6.2.
 
-- a key generation function **KEM.keygen** taking as input a security level and returning a key pair (private key and the associated public key) for this security level.
-- an encapsulation function **KEM.encaps** taking a public key as input and returning a random session key and a ciphertext that is an encapsulation of the session key.
-- a decaspulation function **KEM.decaps** taking as input a private key and a ciphertext and returning a session key.
+> Decapsulate(): ML-KEM.Decaps() from section 6.3.
 
-### KDF
+All security levels of ML-KEM use SHA3-256, SHA3-512, SHAKE256, and SHAKE512 internally. This informs the choice of KDF within this document.
 
-A key derivation function (KDF) is a cryptographic function that deterministically derives one or more secret keys from a secret value using a pseudorandom function. KDFs can be used to stretch keys into longer keys or to obtain keys of a required format.
+<!-- End of ML-KEM section -->
 
-If the session key obtained from the KEM algorithm is long enough to fit into the WRAP algorithm, then the KDF could be equal to the identity function.
+## CMS KEMRecipientInfo Processing Summary {#sec-intro-kemri}
 
-### WRAP
+To support the ML-KEM algorithm, the CMS originator MUST implement Encapsulate().
 
-A wrapping algorithm is a symmetric algorithm protecting data in confidentiality and integrity. It is especially designed to transport key material. the WRAP algorithm consists of two functions :
+Given a content-encryption key CEK, the ML-KEM Algorithm processing by the originator to produce the values that are carried in the CMS KEMRecipientInfo can be summarized as:
 
-- a wrapping function **Wrap** taking a wrapping key and a plaintext key as input and returning a wrapped key.
-- a decaspulation function **Unwrap** taking as input a wrapping key and a wraped key and returning the plaintext key.
+>
+1\. Obtain the shared secret using the Encapsulate() function of the ML-KEM algorithm and the recipient's ML-KEM public key:
 
-In the following, *kekLen* denotes the length in bytes of the wrapping key for the underlying symmetric key-wrapping scheme.
-
-In this scheme, the length of the keying data to be transported MUST be among the lengths supported by the underlying symmetric key-wrapping scheme.
+~~~
+       (ct, ss) = Encapsulate(pk)
+~~~
 
-<!-- End of underlying-components section -->
+>
+2\. Derive a key-encryption key KEK from the shared secret:
```

Comment:
I find this section confusing because it is conceptually crossing the boundaries of the processing done in this document and in the KEMRI document.

I think I would remove this section in its entirety so that casual readers don't think that ML-KEM needs to include a KDF and a WRAP.

Reply:
@ounsworth, this was stolen directly from section 1.3 of rfc5990bis, does the same comment apply there? :) https://www.ietf.org/archive/id/draft-ietf-lamps-rfc5990bis-04.html#name-cms-kemrecipientinfo-proces

There is useful normative language in here:
- CMS originator MUST implement Encapsulate()
- CMS recipient MUST implement Decapsulate()
- KDF MAY be different than the internal ML-KEM one.
Although the first two may be obvious and the third may be more important for rfc5990bis because that document also describes the RSA-KEM process.

If we kept this text, would it be more clear if at the beginning it made clear that these are the same steps from cms-kemri?  Or do you still think it's better to just remove it?

Reply:
Ugg, yeah, that really is copied directly from the KEMRecipientInfo section of 5990bis. 
Ugg, ok, fine then.